docs: added webhook retries guide (#72)
frain-dev/docs: webhook-guides/webhook-retries.mdx

Here’s the configuration for each retry strategy compared:

| Strategy | Formula |
| --- | --- |
| Linear Retry | + 1 hour |
| Exponential Backoff | `(delaySeconds * 2^attempt) + jitter` |
| Exponential Backoff with Capped Duration | `min(delaySeconds * 2^attempt, maxRetrySeconds) + jitter` |

With the following parameters

- **Base:** 20 seconds
- **Jitter: +/-** 10%
- **Delay Growth**: `delaySeconds * 2^attempt`
- **Retry Limit:** 20

To provide additional context, the formula is defined as follows:

- **Delay Growth**: The delay doubles with each retry attempt, allowing for a rapid increase in wait time for successive failures (`delaySeconds * 2^attempt`).
- **Capping**: The new strategy incorporates a default (but configurable) cap of **7200 seconds (**`maxRetrySeconds`) to prevent indefinitely long wait times, thus ensuring the system remains responsive and efficient.
- **Jitter**: A `jitter` value is added to each calculated wait time, providing an additional layer of randomness that further helps distribute the retry attempts over time.

> We wrote an easy script to run these comparison [here](https://github.com/subomi/compare-retries). Feel free to tweak the input and analyse the output.

Reviewer: ```suggestion
We wrote a script to run these comparisons [here](https://github.com/subomi/compare-retries). Feel free to tweak the input and analyze the output.
```

> We wrote an easy script to run these comparison [here](https://github.com/subomi/compare-retries). Feel free to tweak the input and analyse the output.

Reviewer 2: ```suggestion
We wrote a script to run these comparisons [here](https://github.com/subomi/compare-retries). Feel free to tweak the input and analyse the output.
```

### The Results

<Frame caption="A graph comparing the cumulative retry duration for two retry strategies">
  <img src="/images/retries_plot.png" />
</Frame>

Given our requirements above, let's evaluate the two retry strategies shown in the graph based on these criteria:

|  | Linear Retry | Exponential Backoff with Capped Duration |
| --- | --- | --- |
| **Requirement 1** <br /> <br /> The first 5 retries should happen in less than 30 minutes | The graph shows that the linear retry strategy takes about 18,000 seconds by the 5th retry. **18,000 seconds** is approximately **5 hours**, which exceeds the 30-minute limit for the first 5 retries.  | The graph shows that the capped duration strategy takes about 601 seconds by the 5th retry. **601 seconds** is approximately **10 minutes and 1 second**, which doesn’t exceed the 30-minute limit for the first 5 retries.  |

> | **Requirement 2** <br /> <br /> The maximum retry should happen not greater than 24 hours | By the 20th retry attempt, the linear strategy takes about **72,000 seconds**, which translates to **20 hours**, which doesn’t exceed the 24 hours rule. | By the 20th retry attempt the capped duration strategy takes about **89559 seconds**, which translates to **24 hours 52 minutes** which slightly exceeds the 24 hours rule. |

Reviewer 2: ```suggestion
| **Requirement 2** <br /> <br /> The maximum retry should happen not greater than 24 hours | By the 20th retry attempt, the linear strategy takes about **72,000 seconds**, about **20 hours**, which doesn’t exceed the 24 hours rule. | By the 20th retry attempt the capped duration strategy takes about **89559 seconds**, which translates to **24 hours 52 minutes** which slightly exceeds the 24 hours rule. |
```
